From: Simon McVittie <smcv@debian.org>
Date: Wed, 17 Aug 2022 16:47:49 +0100
Subject: tests: Give mutter-test-runner the RUNPATH to find Clutter

Until recently, mutter-test-runner called into libraries that
indirectly depend on (mutter's fork of) Clutter, but did not actually
call into Clutter itself. Commit 1bf70334 "tests/runner: Make test
runner use the headless backend" gave it a direct call into Clutter,
which means the runtime linker will fail unless the executable's
RUNPATH is sufficient to find Clutter.

For future-proofing, do the same for the other test executables.

Resolves: https://gitlab.gnome.org/GNOME/mutter/-/issues/2389
Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2580
Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2581
Signed-off-by: Simon McVittie <smcv@debian.org>
---
 src/tests/meson.build | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/tests/meson.build b/src/tests/meson.build
index ae8c0ca..c68c43a 100644
--- a/src/tests/meson.build
+++ b/src/tests/meson.build
@@ -119,6 +119,7 @@ test_client = executable('mutter-test-client',
   ],
   install: have_installed_tests,
   install_dir: mutter_installed_tests_libexecdir,
+  install_rpath: pkglibdir,
 )
 
 test_runner = executable('mutter-test-runner',
@@ -133,6 +134,7 @@ test_runner = executable('mutter-test-runner',
   dependencies: libmutter_test_dep,
   install: have_installed_tests,
   install_dir: mutter_installed_tests_libexecdir,
+  install_rpath: pkglibdir,
 )
 
 meta_dbus_runner = find_program('meta-dbus-runner.py')
@@ -221,6 +223,7 @@ if have_native_tests
     ],
     install: have_installed_tests,
     install_dir: mutter_installed_tests_libexecdir,
+    install_rpath: pkglibdir,
   )
 
   # Native backend tests
@@ -315,6 +318,7 @@ if have_native_tests
       dependencies: libmutter_test_dep,
       install: have_installed_tests,
       install_dir: mutter_installed_tests_libexecdir,
+      install_rpath: pkglibdir,
     )
 
     test(test_case['name'], test_executable,
@@ -382,6 +386,7 @@ if have_kvm_tests or have_tty_tests
       dependencies: libmutter_test_dep,
       install: have_installed_tests,
       install_dir: mutter_installed_tests_libexecdir,
+      install_rpath: pkglibdir,
     )
     privileged_tests += [
       {
